src/main/scala/nl.knaw.dans.easy.solr4files/EasyUpdateSolr4filesIndexServlet.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files
+
+import nl.knaw.dans.lib.error._
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.scalatra._
+
+import scala.util.Try
+
+class EasyUpdateSolr4filesIndexServlet(app: EasyUpdateSolr4filesIndexApp) extends ScalatraServlet with DebugEnhancedLogging {
+  logger.info("File index Servlet running...")
+
+  get("/") {
+    contentType = "text/plain"
+    Ok("EASY File Index is running.")
+  }
+
+  private def respond(result: Try[String]): ActionResult = {
+    result.map(Ok(_))
+      .doIfFailure { case e => logger.error(e.getMessage, e) }
+      .getOrRecover(_ => InternalServerError())
+  }
+
+  post("/update/:store/:uuid") {
+    respond(app.update(params("store"),params("uuid")))
```

Comment:
What if the store doesn't exist? What if the uuid is invalid? Etc. That's why you need tests! Look at the easy-bag-store servlets for examples.

Reply:
Btw. Same for the other ones